sdks/python/apache_beam/pvalue.py

```diff
@@ -82,7 +82,7 @@ class PValue(object):
   """
 
   def __init__(self,
-               pipeline,  # type: Pipeline
+               pipeline,  # type: Optional[Pipeline]
```

Comment:
Yes, here:

```python
class PCollection(PValue, Generic[T]):
  [...]
  @staticmethod
  def from_runner_api(proto, context):
    # type: (beam_runner_api_pb2.PCollection, PipelineContext) -> PCollection
    # Producer and tag will be filled in later, the key point is that the
    # same object is returned for the same pcollection id.
    return PCollection(
        None,
        element_type=context.element_type_from_coder_id(proto.coder_id),
        windowing=context.windowing_strategies.get_by_id(
            proto.windowing_strategy_id),
        is_bounded=proto.is_bounded == beam_runner_api_pb2.IsBounded.BOUNDED)
```

It would be nice if we could get rid of this case somehow, because by making this optional we have to deal with the possibility of `pvalue.pipeline` being `None` throughout the code base.  I went back and forth on whether to make the arg optional or simply ignore the error in the method above, but I think I decided that the method above was a common case and thus we needed the protection against None-values throughout the code.